### PR TITLE
Test: E2E  Create / Modify / Delete test.

### PR DESCRIPTION
**Pull request by Ealianis**
### Description of your changes
Added new test that will do a create / modify / delete of manifests in a single work update.

I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: tests/e2e/apply_test.go [hunk 1]
```diff
@@ -18,14 +18,14 @@ package e2e
 
 import (
 	"context"
+	"k8s.io/apimachinery/pkg/runtime/schema"
```
Comment (aagusuab) [inline tests/e2e/apply_test.go:21]: The position of this import was changed incorrectly

File: tests/e2e/apply_test.go [hunk 2]
```diff
@@ -199,6 +199,81 @@ var (
...
+				var namespaceToDelete string
+
+				It("should create a secret, modify the existing configmap, and remove the second configmap, from within the spoke", func() {
+					By("getting the existing Work resource on the hub")
+					Eventually(func() error {
+						createdWork, err = retrieveWork(createdWork.Namespace, createdWork.Name)
+						return err
+					}, eventuallyTimeout, eventuallyInterval).ShouldNot(HaveOccurred())
+
+					By("removing the test-namespace manifest")
```
Comment (aagusuab) [inline tests/e2e/apply_test.go:219]: Should this By be used such that 
By("By statement", func() {
})
To show what is included within this by statement?

File: tests/e2e/testmanifests/test-secret.yaml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: test-secret
+  namespace: default
+data:
+  somekey: Q2xpZW50SWQ6IDUxOTEwNTY4LTM0YzktNGQ0ZS1iODA1LTNmNTY3NWQyMDdiYwpDbGllbnRTZWNyZXQ6IDZSLThRfkJvSDNNYm1+eGJpaDhmNVZibHBkWGxzeGQyRnp+WXhjWjYKVGVuYW50SWQ6IDcyZjk4OGJmLTg2ZjEtNDFhZi05MWFiLTJkN2NkMDExZGI0NwpTdWJzY3JpcHRpb25JZDogMmIwM2JmYjgtZTg4NS00NTY2LWE2MmEtOTA5YTExZDcxNjkyClJlc291cmNlR3JvdXA6IGNhcmF2ZWwtZGVtbw==
```
Comment (aagusuab) [inline tests/e2e/testmanifests/test-secret.yaml:7]: Is the key mocked, or is it an actual key?

Comment (Ealianis) [reply tests/e2e/testmanifests/test-secret.yaml:7]: It's not relevant to the test.

File: tests/e2e/apply_test.go (hunk 2 above)
Comment (Ealianis) [reply tests/e2e/apply_test.go:219]: Yes, but I need to refactor the whole file in future PR.

**Pull request merged**